src/uct/ib/ud/base/ud_ep.c

```diff
@@ -685,11 +685,10 @@ void uct_ud_ep_process_rx(uct_ud_iface_t *iface, uct_ud_neth_t *neth, unsigned b
         ucs_queue_push(&iface->rx.pending_q, &skb->u.am.queue);
     } else {
         /* Avoid reordering with respect to pending operations, if user AM handler
-         * initiates sends. This flag would be cleared after all incoming messages
+         * initiates sends from any endpoint created on the iface.
+         * This flag would be cleared after all incoming messages
          * are processed. */
-        if (ucs_unlikely(!ucs_arbiter_group_is_empty(&ep->tx.pending.group))) {
-            uct_ud_iface_raise_pending_async_ev(iface);
-        }
+        uct_ud_iface_raise_pending_async_ev(iface);
```

Comment:
this mechanism is broken, since user am handler can initiate sends even on another iface (multi-rail), let's think how to solve it

Reply:
I'm not sure if the issue is actual for multi-lane. This bug occurs from **sync** progress because pending is placed after RX polling where user callback is called. Another iface in m-lane case has own progress, pending queue, etc. So sending from callback on EP from other iface should be blocked by different mechanism if it has pending requests.
Do you see exact scenario where this mechanism is broken?